### DISCUSSION TITLE
How to use in Frame for iOS?

### DISCUSSION THREAD

**Discussion by RbBtSn0w** [emma-k-alexandra/RTree#8] [Q&A] [answered]
The base protocol is PointN.

for iOS slide, have a space zone by frame: CGRect

So how define it?

**Comment by emma-k-alexandra** [comment, marked as answer]
To use a `CGRect`, you'd need to implement `SpatialObject` conformance for `CGRect`. For your `SpatialObject.Point`, you would could use a 2D Vector, `Point2D` as described in the Readme, storing two 2D vectors for the top left and bottom right points of your CGRect.

Then, your `minimumBoundingRectangle` would just be the top left and bottom right points of your CGRect. For `distanceSquared`, you'd need to find the shortest distance from your CGRect to the given point.

I don't know what your use case is, but if you're working with CGRects you might be working with visual elements on screen. Thus, your data is changing at run time. This package stores data on disk, which might not be the best solution for you. If you're generating and using these trees at runtime, you might be better suited by Apple's [GKRTree](https://developer.apple.com/documentation/gameplaykit/gkrtree) in GameplayKit.

**Reply by RbBtSn0w** [reply]
Thanks.

GKRTree is limited above on iOS10.My project is working on iOS9.

My scene looks like this.

All Indexes in the CollectionView correspond to a dynamic frame (CGRect), when the ScrollView scrolls to a certain area of the Rect range quickly returns to the existing Index of the current screen.

The short answer is to sort the frame by position size by quick sorting, and then through a loop, the screen is found inside the return.

But there is an efficiency problem in doing so, because the uncertainty of the quick order, as well as a full loop, slows down the execution speed.

That's why I thought of using RTree to solve the above search, and after the frame is updated, RTree can automatically complete the update of their respective nodes internally, and when searching again, it can also have efficient performance.
